drafts/arcalinea-tx-expiry/draft.rst

```diff
@@ -0,0 +1,76 @@
+**Title:** Transaction Expiry
+
+**Author:** Jay Graber
+
+**Status:** Active
+
+**Category:** Standards
+
+**Created:** 2018-01-09
+
+Abstract
+===========
+
+This is an Standards ZIP describing a new consensus rule to set an expiration time for transactions that are not mined to be removed from the mempool.
+
+Motivation
+===========
+
+Transactions that have insufficient fees are often not mined. This indeterminism is a source of confusion for users and wallets. Allowing transactions to set a block height after which it expires from the mempool would provide certainty around how long a transaction has to confirm before it is rejected by the network and must be re-sent.
+
+Advantages include optimizing mempool performance by removing transactions unlikely to be mined, and potentially simplifying bidirectional payment channels by reducing the need to store and compress revocations for past states, since transactions not committed to the chain could expire and become invalid after a period of time.
+
+If the expiry is block height N, it means the transaction must have either: 1. have been included in N-1, or 2. be included in block N. N+1 will be too late, and the transaction will be removed from the mempool.
```

Comment:
"N+1 will be too late, and the transaction will be removed from the mempool" is the recommended behavior for nodes and miners, but the ZIP is still missing a clear specification of the *consensus rule*, like "the transaction will not be considered valid if included in block of depth greater than N".